flow/embedded_views.h

```diff
@@ -17,7 +17,24 @@
 
 namespace flutter {
 
-enum MutatorType { clip_rect, clip_rrect, clip_path, transform };
+enum MutatorType { clip_rect, clip_rrect, clip_path, transform, opacity };
+
+// Stores information required for an opacity Mutator.
+// Matches the members in a `OpacityLayer`.
+struct OpacityParams {
+  std::reference_wrapper<int> alpha;
+  std::reference_wrapper<SkPoint> offset;
```

Comment:
Haven't read the rest of the PR yet, but looking at this I'm not sure what does an offset means for opacity, should probably add a comment.